sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/SearchClientBuilder.java

```diff
@@ -98,24 +108,77 @@ public SearchClient buildClient() {
      * other builder settings are ignored.
      *
      * @return A SearchClient with the options set from the builder.
-     * @throws NullPointerException If {@code indexName} or {@code endpoint} are {@code null}.
+     * @throws NullPointerException If {@code indexName} or {@code endpoint} are null.
      */
     public SearchAsyncClient buildAsyncClient() {
-        Objects.requireNonNull(indexName, "'indexName' cannot be null.");
-        Objects.requireNonNull(endpoint, "'endpoint' cannot be null.");
+        validateIndexNameAndEndpoint();
         SearchServiceVersion buildVersion = (serviceVersion == null)
             ? SearchServiceVersion.getLatest()
             : serviceVersion;
 
+        HttpPipeline pipeline = getHttpPipeline();
+        return new SearchAsyncClient(endpoint, indexName, buildVersion, pipeline, jsonSerializer,
+            buildRestClient(pipeline));
+    }
+
+    /**
+     * Creates a {@link SearchIndexingBufferedSender} based on options set in the builder. Every time this is called a
+     * new instance of {@link SearchIndexingBufferedSender} is created.
+     * <p>
+     * If {@link #pipeline(HttpPipeline) pipeline} is set, then only the {@code pipeline}, {@link #endpoint(String)
+     * endpoint}, and {@link #indexName(String) indexName} are used to create the {@link SearchIndexingBufferedSender
+     * client}. All other builder settings are ignored.
+     *
+     * @param options Batching configurations for the client.
+     * @param <T> Document type for the client.
+     * @return A SearchIndexingBufferedSender with the options set from the builder.
+     * @throws NullPointerException If {@code indexName}, {@code endpoint}, or {@code options} are null.
+     */
+    public <T> SearchIndexingBufferedSender<T> buildBufferedSender(SearchIndexingBufferedSenderOptions<T> options) {
+        return new SearchIndexingBufferedSender<>(buildBufferedAsyncSender(options));
+    }
+
+    /**
+     * Creates a {@link SearchIndexingBufferedAsyncSender} based on options set in the builder. Every time this is
+     * called a new instance of {@link SearchIndexingBufferedAsyncSender} is created.
+     * <p>
+     * If {@link #pipeline(HttpPipeline) pipeline} is set, then only the {@code pipeline}, {@link #endpoint(String)
+     * endpoint}, and {@link #indexName(String) indexName} are used to create the {@link
+     * SearchIndexingBufferedAsyncSender client}. All other builder settings are ignored.
+     *
+     * @param options Batching configurations for the client.
+     * @param <T> Document type for the client.
+     * @return A SearchIndexingBufferedAsyncSender with the options set from the builder.
+     * @throws NullPointerException If {@code indexName}, {@code endpoint}, or {@code options} are null.
```

Comment:
Why can't the options be nullable here?

Reply:
The options bag has a required configuration for the indexing client to work, so if it is null the application won't run properly

Reply:
I'm not a huge fan of the options being required. I've pushed back in the past on builders requiring fluent options types, and especially in this case it feels odd because this options type only appears to be used as part of this builder. Have you considered whether this is a case where the branching builder pattern should be used instead?